eogrow/pipelines/download_batch.py

```diff
@@ -153,10 +187,56 @@ def _create_new_batch_request(self) -> BatchRequest:
                 resolution=self.config.area.resolution,
                 buffer=(self.config.area.tile_buffer, self.config.area.tile_buffer),
             ),
-            output=SentinelHubBatch.output(default_tile_path=f"{data_folder}/<tileName>/<outputId>.<format>"),
+            output=SentinelHubBatch.output(
+                default_tile_path=f"{data_folder}/<tileName>/<outputId>.<format>", **self.config.batch_output_kwargs
+            ),
             description=f"eo-grow - {self.__class__.__name__} pipeline with ID {self.pipeline_id}",
         )
 
+    def _trigger_user_action(self, batch_request: BatchRequest) -> BatchUserAction:
+        """According to status and configuration parameters decide what kind of user action to perform."""
+        if self.config.analysis_only:
+            if batch_request.status is BatchRequestStatus.CREATED:
+                self.batch_client.start_analysis(batch_request)
+                LOGGER.info("Triggered batch job analysis.")
+                return BatchUserAction.ANALYSE
+
+            LOGGER.info(
+                "Didn't trigger analysis because current batch request status is %s.", batch_request.status.value
+            )
+            return BatchUserAction.NONE
+
+        if batch_request.status in [
+            BatchRequestStatus.CREATED,
+            BatchRequestStatus.ANALYSING,
+            BatchRequestStatus.ANALYSIS_DONE,
+        ]:
+            self.batch_client.start_job(batch_request)
+            LOGGER.info("Started running batch job.")
+            return BatchUserAction.START
+
+        if batch_request.status is BatchRequestStatus.PARTIAL:
+            self.batch_client.restart_job(batch_request)
+            LOGGER.info("Restarted partially failed batch job.")
+            return BatchUserAction.START
+
+        LOGGER.info("Didn't trigger batch job because current batch request status is %s", batch_request.status.value)
+        return BatchUserAction.NONE
+
+    def ensure_batch_grid(self, request_id: str) -> None:
+        """This method ensures that area manager caches batch grid into the storage."""
+        self.area_manager = cast(BatchAreaManager, self.area_manager)
```

Comment:
Just wondering if we would perhaps even REQUIRE that the area manager is  a `BatchAreaManager` or a sublcass of it? :thinking:

Reply:
It kinda is required because we rely on the fact that area manager has attribute `batch_id`. Technically, one could write an independent area manager with this attribute and it could still work. But in such case I would recommend to just subclass `BatchAreaManager`.

So for now, I just added casting to make mypy happy. But if you think, I can also add a check in init that makes sure that `self.area_manager` is an instance of `BatchAreaManager`.

Reply:
Since we're relying on that, I'd say that maybe we really should check that somewhere in the `__init__`.

Reply:
Done.